server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -665,8 +665,13 @@ public final DocumentParserContext createNestedContext(NestedObjectMapper nested
         if (idField != null) {
             // We just need to store the id as indexed field, so that IndexWriter#deleteDocuments(term) can then
             // delete it when the root document is deleted too.
-            // NOTE: we don't support nested fields in tsdb so it's safe to assume the standard id mapper.
             doc.add(new StringField(IdFieldMapper.NAME, idField.binaryValue(), Field.Store.NO));
+        } else if (indexSettings().getMode() == IndexMode.TIME_SERIES) {
+            // For time series indices, the _id is generated from the _tsid, which in turn is generated from the values of the configured
```

Comment:
Maybe we should add an assert that `getRoutingFields()` doesn't return a reference to `RoutingFields.Noop#INSTANCE`? Just to make sure we are able to collect dimension values in order to generate _tsid / _id at a later stage?